code/game/objects/items/mail.dm

```diff
@@ -291,12 +291,24 @@
 		info = "<i>You need to escape the simulation. Don't forget the numbers, they help you remember:</i> '[rand(0,9)][rand(0,9)][rand(0,9)]...'"
 		return
 	var/code = random_nukecode()
+	var/code_already_set = FALSE
 	for(var/obj/machinery/nuclearbomb/selfdestruct/self_destruct in GLOB.nuke_list)
+		if(self_destruct.r_code != "ADMIN")
+			code_already_set = TRUE
+			code = self_destruct.r_code
+			break
 		self_destruct.r_code = code
```

Comment:
if you had multiple self destruct terminals, this could bug out and leave a nuke with a different code from the rest.